### PR TITLE
Filter movies by genre and title.

### PR DESCRIPTION
**Pull request by j1nma**
## Summary
* I followed the "Búsqueda y filtrado" card to add the possibility of filtering the list according to movie genre and title.

## Trello Card Link
https://trello.com/c/Ovwbi6RL

### A user can search and filter on the system's movies.
image

image

### CODE REVIEW

File: src/Controllers/MoviesController.cs [hunk 1]
```diff
@@ -21,19 +24,37 @@ public IUnitOfWork UnitOfWork
...
+            var genres = _unitOfWork.MovieRepository.GetGenres();
+
             var movies = _unitOfWork.MovieRepository.GetAll();
 
-            var moviesModels = new List<MovieViewModel>();
+            if (!String.IsNullOrEmpty(searchString))
+            {
+                movies = movies.Where(s => s.Title.Contains(searchString));
+            }
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -21,19 +24,37 @@ public IUnitOfWork UnitOfWork
...
+            if (!String.IsNullOrEmpty(movieGenre))
+            {
+                movies = movies.Where(x => x.Genre == movieGenre);
+            }
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs:40]: i think this is not working well. you are filtering first by title and saving the result in a variable called 'movies' and then filtering by genre and saving in that variable too.

File: src/Controllers/MoviesController.cs [hunk 3]
```diff
@@ -21,19 +24,37 @@ public IUnitOfWork UnitOfWork
...
+                SelectListItem selectListItem = new SelectListItem() { Text = m.Genre, Value = m.Genre };
+                if (!movieGenreVM.genres.Any(l => l.Value == selectListItem.Value))
+                {
+                    movieGenreVM.genres.Add(selectListItem);
+                }
             }
 
-            return View(moviesModels);
+            movieGenreVM.genres = movieGenreVM.genres.Distinct().ToList();
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 4]
```diff
@@ -159,7 +180,7 @@ public IActionResult DeleteConfirmed(int id)
 
             _unitOfWork.MovieRepository.Remove(movie);
             _unitOfWork.Complete();
-            
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs:181]: remove this empty line

**Review by gzamudio — CHANGES_REQUESTED** [review]

File: src/Controllers/MoviesController.cs (hunk 2 above)
Comment (j1nma) [reply src/Controllers/MoviesController.cs:40]: This way I can filter by title, genre or both. I could put both logics of NullOrEmpty inside a single Where(), but that would not look nice.

Comment (gzamudio) [reply src/Controllers/MoviesController.cs:40]: ok! yes, i understand now

**Review by gzamudio — APPROVED** [review]

**Pull request merged**